services/sidecar/rollup/derivation/batch_builder.go

```diff
@@ -7,32 +7,32 @@ import (
 	"math/big"
 
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
 	"github.com/specularL2/specular/services/sidecar/rollup/types"
 	"github.com/specularL2/specular/services/sidecar/utils/log"
 )
 
 type batchBuilder struct {
 	maxBatchSize uint64
 
-	pendingBlocks   []DerivationBlock
-	builtBatchAttrs *BatchAttributes
+	pendingBlocks  []DerivationBlock
+	builtBatchData *[]byte
```

Comment:
dont do `*[]byte`, just use `[]byte`.

Reply:
coming from C++ i've been confused about whether to pass large variables by "value" or "pointer".

just did some research and seems like the internet's consensus is that parameters in go should only be "passed-by-pointer" if you wish to allow the function to modify the value passed to it. Because, escape analysis allows go to optimize "pass-by-value". does this sound right?